### PR TITLE
[Multilane] Adds start and end Endpoint computation to Connections

### PR DESCRIPTION
**Pull request by agalbachicar**
This PR is a follow up of #7325 as part of issue #7194. It adds methods to `Connection` class to compute lane `Endpoint`s among other refactors. 

To sum up,  this PR includes:

- `Connection` constructor signatures are modified and do not need the end `Endpoint` anymore. Just end `EndpointZ` is needed. By doing this, we avoid the possibility of building a `Connection` that does not connect _start_ and _end_ `Endpoints`.
- `Connection`s build a `RoadCurve` at construction time and provide an API to construct a new one as requested.
- `Connection` will be in charge of building the elevation and superelevation polynomials for its `RoadCurve`. Tests are also included to check the creation.
- `Lane`-centerline offset computation is moved to `Connection` class.
- Provides `Connection::LaneStart()` and `Connection::LaneEnd()` methods to allow query lane _start_ and _end_ `Endpoint`s.
- Provides several test conditions with different geometries in order to validate new methods.
- Modifies `Builder`'s implementation to support previous changes.

Future PRs will modify loader methods and YAML format description as #4934 explains.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/7343)

### CODE REVIEW

**Comment by agalbachicar** [comment]
+@maddog-tri for feature review please.

---

Review status: 0 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-KxtKhN6WFL4YJumC9Xc:b-5klkw7)*

**Comment by jamiesnape** [comment]
@drake-jenkins-bot linux-xenial-clang-cmake-experimental please
@drake-jenkins-bot linux-xenial-gcc-cmake-experimental-matlab please

**Comment by maddog-tri** [comment]
Just to be sure:  "and includes addresses pending items to finish issue #7194" means to say "and includes more items necessary for #7194".  This PR is not intended to _finish_ #7194, right?  Because... the `Builder::Connect()` methods still need to be modified to achieve both lane-based and reference-line-base connection verbs (with alignment of arbitrary lane indices).

---

Reviewed 1 of 5 files at r1.
Review status: 1 of 5 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-KxyesnBs93ZeOpcvPjE:bbyo1xm)*

**Comment by maddog-tri** [comment]
First pass complete; PTAL.

---

Reviewed 3 of 5 files at r1.
Review status: 4 of 5 files reviewed at latest revision, 8 unresolved discussions.

---

*[drake/automotive/maliput/multilane/connection.cc, line 108 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxyqM6W9_6cL7MZwA1l:-KxyqM6W9_6cL7MZwA1m:b-p3embh) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.cc#L108)):*
> ```cpp
>                                   road_curve_->elevation().f_dot_p(0.));
>   return Endpoint(EndpointXy(position[0], position[1], rotation.yaw()),
>                   EndpointZ(position[2], w_prime.z(), start_.z().theta(),
> ```

`w_prime.z()` is dz/dp... but `z_dot` is an actual, physical "rise over run" slope of the road --- in other words, "dz/d(length-of-constant-r-curve-projected-on-the-xy-plane)". 

So... it needs to be scaled by "dp/d(blah)" --- e.g., for a line-connection, `1 / line_length_`; for an arc-connection... the reciprocal of the path-length of an arc with the effective radius of the lane... after accounting for superelevation.

(Hmm... I wonder if derivatives of superelevation also affect this....)

---

*[drake/automotive/maliput/multilane/connection.cc, line 121 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-Ky-cSschilGkP3AzV0q:-Ky-cSschilGkP3AzV0r:b-232k60) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.cc#L121)):*
> ```cpp
>                                   road_curve_->elevation().f_dot_p(1.));
>   return Endpoint(EndpointXy(position[0], position[1], rotation.yaw()),
>                   EndpointZ(position[2], w_prime.z(), start_.z().theta(),
> ```

See comment above about `w_prime.z()` vs `z_dot`.

---

*[drake/automotive/maliput/multilane/connection.h, line 189 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxyiTZ8B-9U-hWFz3Od:-KxyiTZ9lWmx_a_Hfz_6:b8ou1g) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L189)):*
> ```objc
>   /// Constructs a line-segment connection.
>   ///
>   /// It begins at `start` and extends on the plane z=0 `line_length` distance
> ```

Need to be more specific about what "It" is --- now it is the planar reference curve for the Segment (which is no longer necessarily the centerline of a single lane).

---

*[drake/automotive/maliput/multilane/connection.h, line 210 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxyjSuu6t2349ZCvMJx:-KxyjSuvvDReraA4o_S8:b-6sq4us) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L210)):*
> ```objc
>   /// Constructs an arc-segment connection.
>   ///
>   /// It begins at `start` and extends on the plane z=0 with
> ```

Same as above: need to be specific about what "It" is.

---

*[drake/automotive/maliput/multilane/connection.h, line 211 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxyjdldcmbqTVkNXzl0:-KxyjdldcmbqTVkNXzl1:b2342po) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L211)):*
> ```objc
>   ///
>   /// It begins at `start` and extends on the plane z=0 with
>   /// `arc_offset.radius()` an angle span of `arc_offset.d_theta()`. `end_z`
> ```

"an" --> "and"?

---

*[drake/automotive/maliput/multilane/connection.h, line 296 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxykomrXFYbIUw-RUin:-KxykomrXFYbIUw-RUio:bxx97a3) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L296)):*
> ```objc
> 
>   /// Computes the location and heading of a `lane_index` Lane at its beginning
>   /// creating an Endpoint with that information.
> ```

It's more than location and heading (that's just EndpointXy --- then there is the EndpointZ parts, too).  Maybe better to just phrase as: "Returns an Endpoint describing the start of the `lane_index` lane."

---

*[drake/automotive/maliput/multilane/connection.h, line 300 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxylJSOlbhyPHkMT927:-KxylJSOlbhyPHkMT928:bj0g7r6) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L300)):*
> ```objc
> 
>   /// Computes the location and heading of a `lane_index` Lane at its end
>   /// creating an Endpoint with that information.
> ```

(Same comment as above.)

---

*[drake/automotive/maliput/multilane/test/multilane_connection_test.cc, line 329 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-Ky-d5T4eCIGBqH6BuWW:-Ky-d5T4eCIGBqH6BuWX:bwtfqyj) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/test/multilane_connection_test.cc#L329)):*
> ```cpp
> // elevation polynomials.
> class MultilaneConnectionElevationEndpointTest
>     : public ::testing::TestWithParam<CubicPolynomial> {
> ```

It is confusing to use `CubicPolynomial` as a container for test parameters, because the test parameters `CubicPolynomial` has nothing to do with the cubic polynomials constructed by the `Connection` logic.  (Also, it obfuscates the problem that `f_dot_p()` is constant in all the test cases, and thus `start_z.z_dot == end_z.z_dot` in all cases (and... `start_z.z == end_z.z` in the superelevation tests).)

It would be better to create a struct which very explicitly contains the required parameters, e.g.,
```
struct Parameters {
  EndpointZ start_z;
  EndpointZ end_z;
  int num_lanes;
  double r0;
};
```
...and also consolidate the "SuperelevationEndpointTest" with "ElevationEndpointTest", because they appear identical except for the hard-coded `kNumLanes` and `kR0` constants.

Then it will also be clear that, except for variations due to lane offsets, the results of `LaneStart()` and `LaneEnd()` should really match the start/end values used to construct the dut.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-Ky-iYaw5pHOKaAO8TgD:b-2o5461)*

**Comment by agalbachicar** [comment]
Thanks for the clarification. I'm going to change #7194 description so as to reflect the missing API changes and I've modified this PR's description.

PTAL.

---

Review status: 2 of 5 files reviewed at latest revision, 8 unresolved discussions.

---

*[drake/automotive/maliput/multilane/connection.cc, line 108 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxyqM6W9_6cL7MZwA1l:-Ky2UXT0WX2eQCRbGDgG:bmpx1eo) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.cc#L108)):*
<details><summary><i>Previously, maddog-tri (Matt Marjanović) wrote…</i></summary><blockquote>

`w_prime.z()` is dz/dp... but `z_dot` is an actual, physical "rise over run" slope of the road --- in other words, "dz/d(length-of-constant-r-curve-projected-on-the-xy-plane)". 

So... it needs to be scaled by "dp/d(blah)" --- e.g., for a line-connection, `1 / line_length_`; for an arc-connection... the reciprocal of the path-length of an arc with the effective radius of the lane... after accounting for superelevation.

(Hmm... I wonder if derivatives of superelevation also affect this....)
</blockquote></details>

Done.

As far as I understand, `theta_dot` should be scaled too but `theta` shouldn't since it is kept along `r` direction for a given `p`.

---

*[drake/automotive/maliput/multilane/connection.cc, line 121 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-Ky-cSschilGkP3AzV0q:-Ky2UY1kfPwks2NrXQpR:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.cc#L121)):*
<details><summary><i>Previously, maddog-tri (Matt Marjanović) wrote…</i></summary><blockquote>

See comment above about `w_prime.z()` vs `z_dot`.
</blockquote></details>

Done.

---

*[drake/automotive/maliput/multilane/connection.h, line 189 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxyiTZ8B-9U-hWFz3Od:-Ky2VNBsO_eADDem992g:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L189)):*
<details><summary><i>Previously, maddog-tri (Matt Marjanović) wrote…</i></summary><blockquote>

Need to be more specific about what "It" is --- now it is the planar reference curve for the Segment (which is no longer necessarily the centerline of a single lane).
</blockquote></details>

Done.

---

*[drake/automotive/maliput/multilane/connection.h, line 210 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxyjSuu6t2349ZCvMJx:-Ky2VNtK6XBIMzlnI2fo:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L210)):*
<details><summary><i>Previously, maddog-tri (Matt Marjanović) wrote…</i></summary><blockquote>

Same as above: need to be specific about what "It" is.
</blockquote></details>

Done.

---

*[drake/automotive/maliput/multilane/connection.h, line 211 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxyjdldcmbqTVkNXzl0:-Ky1Mpa8e0W_Yp8HyM91:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L211)):*
<details><summary><i>Previously, maddog-tri (Matt Marjanović) wrote…</i></summary><blockquote>

"an" --> "and"?
</blockquote></details>

Done.

---

*[drake/automotive/maliput/multilane/connection.h, line 296 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxykomrXFYbIUw-RUin:-Ky1MmG_K1ByuIv_Aebp:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L296)):*
<details><summary><i>Previously, maddog-tri (Matt Marjanović) wrote…</i></summary><blockquote>

It's more than location and heading (that's just EndpointXy --- then there is the EndpointZ parts, too).  Maybe better to just phrase as: "Returns an Endpoint describing the start of the `lane_index` lane."
</blockquote></details>

Done.

---

*[drake/automotive/maliput/multilane/connection.h, line 300 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxylJSOlbhyPHkMT927:-Ky1Mm3Y-y52b1_ZFL95:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.h#L300)):*
<details><summary><i>Previously, maddog-tri (Matt Marjanović) wrote…</i></summary><blockquote>

(Same comment as above.)
</blockquote></details>

Done.

---

*[drake/automotive/maliput/multilane/test/multilane_connection_test.cc, line 329 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-Ky-d5T4eCIGBqH6BuWW:-KyGsR7NMtpBRcHzTzTI:b-1zwzeg) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/test/multilane_connection_test.cc#L329)):*
<details><summary><i>Previously, maddog-tri (Matt Marjanović) wrote…</i></summary><blockquote>

It is confusing to use `CubicPolynomial` as a container for test parameters, because the test parameters `CubicPolynomial` has nothing to do with the cubic polynomials constructed by the `Connection` logic.  (Also, it obfuscates the problem that `f_dot_p()` is constant in all the test cases, and thus `start_z.z_dot == end_z.z_dot` in all cases (and... `start_z.z == end_z.z` in the superelevation tests).)

It would be better to create a struct which very explicitly contains the required parameters, e.g.,
```
struct Parameters {
  EndpointZ start_z;
  EndpointZ end_z;
  int num_lanes;
  double r0;
};
```
...and also consolidate the "SuperelevationEndpointTest" with "ElevationEndpointTest", because they appear identical except for the hard-coded `kNumLanes` and `kR0` constants.

Then it will also be clear that, except for variations due to lane offsets, the results of `LaneStart()` and `LaneEnd()` should really match the start/end values used to construct the dut.
</blockquote></details>

Done. Thanks for the suggestion, it is much better.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-KyH4rLNWQXxYTKE5Wl2:bldi7f6)*

**Comment by maddog-tri** [comment]
Ok -- second pass complete.

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/automotive/maliput/multilane/connection.cc, line 108 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KxyqM6W9_6cL7MZwA1l:-KyN4Ki4EaEZX1GEc6b_:bqvgmer) ([raw file](https://github.com/robotlocomotion/drake/blob/d286d782c44d87c787068092c69e759fcc350cda/drake/automotive/maliput/multilane/connection.cc#L108)):*
<details><summary><i>Previously, agalbachicar (Agustin Alba Chicar) wrote…</i></summary><blockquote>

Done.

As far as I understand, `theta_dot` should be scaled too but `theta` shouldn't since it is kept along `r` direction for a given `p`.
</blockquote></details>

Good catch on `theta_dot`.

Please add a TODO, also, near "Computes ∂p/∂t based on Connection geometry type.", and in LaneEnd, too:
```
   // TODO(maddog-tri)  A (second-order?) contribution of theta_dot to ∂p/∂t is being ignored.
```

(Imagine theta around 90 degrees, and variation of theta projects into variation on the xy-plane, even for the kLine connection.)

---

*[drake/automotive/maliput/multilane/test/multilane_connection_test.cc, line 228 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyMw8LIndsrS0D6AXLb:-KyMw8LIndsrS0D6AXLc:bm88qt3) ([raw file](https://github.com/robotlocomotion/drake/blob/dd26222bf3c8b780fcbf4b974e53c884430b0fd6/drake/automotive/maliput/multilane/test/multilane_connection_test.cc#L228)):*
> ```cpp
>                                               kEndElevatedEndpoint.z().z()),
>                               complex_end, kVeryExact));
> ```

How were the literal values below derived?  (E.g., if the answer is "Calculated independently in octave.", it would be good to also include the octave expression(s).)

---

*[drake/automotive/maliput/multilane/test/multilane_connection_test.cc, line 311 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyMvgd73DdUvOWL117A:-KyMvgd73DdUvOWL117B:bm88qt3) ([raw file](https://github.com/robotlocomotion/drake/blob/dd26222bf3c8b780fcbf4b974e53c884430b0fd6/drake/automotive/maliput/multilane/test/multilane_connection_test.cc#L311)):*
> ```cpp
>                                               kEndElevatedEndpoint.z().z()),
>                               complex_end, kVeryExact));
> ```

How were the literal values below derived?  (E.g., if the answer is "Calculated independently in octave.", it would be good to also include the octave expression(s).)

---

*[drake/automotive/maliput/multilane/test/multilane_connection_test.cc, line 389 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyN2qDZ8Io26JURAQKU:-KyN2qDZ8Io26JURAQKV:bsway3b) ([raw file](https://github.com/robotlocomotion/drake/blob/dd26222bf3c8b780fcbf4b974e53c884430b0fd6/drake/automotive/maliput/multilane/test/multilane_connection_test.cc#L389)):*
> ```cpp
>     // Start endpoints.
>     const double effective_radius =
>         kRadius - r0 - static_cast<double>(i) * kLaneWidth;
> ```

Probably want a TODO here mentioning that this does not account for non-zero superelevation (which is not tested yet --- i.e., no test case yet with both non-zero z_dot and non-zero superelevation).

OR... better yet, add a test case that exercises that combination.

---

*[drake/automotive/maliput/multilane/test/multilane_connection_test.cc, line 435 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyN3s54HbxOdKI7WKtD:-KyN3s54HbxOdKI7WKtE:bkaub5t) ([raw file](https://github.com/robotlocomotion/drake/blob/dd26222bf3c8b780fcbf4b974e53c884430b0fd6/drake/automotive/maliput/multilane/test/multilane_connection_test.cc#L435)):*
> ```cpp
> }
> 
> INSTANTIATE_TEST_CASE_P(EndpointZ, MultilaneConnectionEndpointZTest,
> ```

Hmm... there are no test cases with non-zero theta_dot, or with all non-zero parameters for at least one EndpointZ.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-KyN6Jc4KJqECOIk8IHO:b-w52xhr)*

**Comment by maddog-tri** [comment]
Great --- it's ready for platform review.

---

Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/maliput/multilane/test/multilane_connection_test.cc, line 435 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyN3s54HbxOdKI7WKtD:-KyXiH-KUsc4jhchBFFo:bdbk24r) ([raw file](https://github.com/robotlocomotion/drake/blob/dd26222bf3c8b780fcbf4b974e53c884430b0fd6/drake/automotive/maliput/multilane/test/multilane_connection_test.cc#L435)):*
<details><summary><i>Previously, agalbachicar (Agustin Alba Chicar) wrote…</i></summary><blockquote>

Done.

Thanks, your suggestion uncovered a bug related to `theta_dot`. It is already expressed in terms of `∂t` but it should be scaled (`∂t_0 / ∂t_1`) because of the r-offset when the planar length changes. 
</blockquote></details>

Yay for tests!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-KyXi_8NtuylyurFwd7L:b4iaccb)*

**Comment by agalbachicar** [comment]
@maddog-tri I don't know if there is something missing but I think I need your lgtm emoji.

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-KypL7jz9VNyw9eiYOqI:ba7bdkc)*

**Comment by agalbachicar** [comment]
+@ericcousineau-tri for platform review please.

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-Kypc7hPsB-prAu6ZflJ:b72l57d)*

**Comment by EricCousineau-TRI** [comment]
:lgtm:, with some minor nits here and there.

---

Reviewed 2 of 5 files at r1, 1 of 3 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/automotive/maliput/multilane/connection.cc, line 49 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqCTHoqZTfR005zGN-:-KyqCTHpC_Lkl4UxTHBY:b-rzmi6) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.cc#L49)):*
> ```cpp
>   DRAKE_DEMAND(line_length_ > 0.);
>   // Computes end Endpoint and RoadCurve.
>   end_ = {{start_.xy().x() + line_length_ * std::cos(start_.xy().heading()),
> ```

BTW minor nit Since `end_` is not being construction-initialized, could you reword this to say `end_ = Endpoint({x, y, th}, z)` to make this operation a little clearer?
(Same for other instances -- this is also just from my own opinion; feel free to reject it.)

---

*[drake/automotive/maliput/multilane/connection.cc, line 118 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqDBcALUng-216cxhQ:-KyqDBcBGWsiTV6Rikvh:busp6qq) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.cc#L118)):*
> ```cpp
>   // Computes ∂p/∂t based on Connection geometry type.
> 
>   // TODO(maddog-tri)  A (second-order?) contribution of theta_dot to ∂p/∂t is
> ```

BTW Just out of curiosity, can I ask what "order" refers to here? Is it of polynomial order?

---

*[drake/automotive/maliput/multilane/connection.cc, line 173 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqDS0CGSdQGGTlFTHq:-KyqDS0DzNDddDTZFj7t:b-hyj0o1) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.cc#L173)):*
> ```cpp
>       end_.z().theta_dot() :
>       end_.z().theta_dot() * std::abs(d_theta_ * radius_) / planar_length;
>   return Endpoint(EndpointXy(position[0], position[1], rotation.yaw()),
> ```

BTW minor nit Inconsistency with above construction `{{x, y, th}, z}`.
I am fine with `Endpoint({x, y, th}, z)`as a construction (or honestly, any of them), but it'd be nice to have one form or the other.

---

*[drake/automotive/maliput/multilane/connection.cc, line 234 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqDwpql1_29lBhsTPg:-KyqDwprA1zcwEcq1-bz:b-j21a1p) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.cc#L234)):*
> ```cpp
>           center, radius_, theta0_, d_theta_, elevation, superelevation);
>     };
>     default: { DRAKE_ABORT(); }
> ```

BTW style nit Could you have this move to the next line?

---

*[drake/automotive/maliput/multilane/connection.h, line 307 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqEKuxq4Y6NE_LXgg3:-KyqEKuxq4Y6NE_LXgg4:b-w0mxg4) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.h#L307)):*
> ```objc
>  private:
>   const Type type_{};
>   const std::string id_{};
> ```

BTW We only really need `{}` member declaration initialization for PODs that would not have it otherwise.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-KyqG0qdMWdUZpLxcpcL:bc63yqn)*

**Comment by agalbachicar** [comment]
I've addressed most of the comments and answered the question. PTAL. 

---

Review status: 3 of 5 files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/automotive/maliput/multilane/connection.cc, line 49 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqCTHoqZTfR005zGN-:-KyqVdushRK7YYEJIX54:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.cc#L49)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

BTW minor nit Since `end_` is not being construction-initialized, could you reword this to say `end_ = Endpoint({x, y, th}, z)` to make this operation a little clearer?
(Same for other instances -- this is also just from my own opinion; feel free to reject it.)
</blockquote></details>

Done.

---

*[drake/automotive/maliput/multilane/connection.cc, line 118 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqDBcALUng-216cxhQ:-KyqY-h-igE2vJHUx0gB:b-p3tk78) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.cc#L118)):*
> hat "order" refers to here? Is it of polynomial order?
Yes, it is. Both elevation and superelevation are polynomials.

---

*[drake/automotive/maliput/multilane/connection.cc, line 173 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqDS0CGSdQGGTlFTHq:-KyqWq5rtdJEeKeu6h-o:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.cc#L173)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

BTW minor nit Inconsistency with above construction `{{x, y, th}, z}`.
I am fine with `Endpoint({x, y, th}, z)`as a construction (or honestly, any of them), but it'd be nice to have one form or the other.
</blockquote></details>

Done.

---

*[drake/automotive/maliput/multilane/connection.cc, line 234 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqDwpql1_29lBhsTPg:-KyqWtgGxfCXMgU4Ir38:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.cc#L234)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

BTW style nit Could you have this move to the next line?
</blockquote></details>

Done.

---

*[drake/automotive/maliput/multilane/connection.h, line 307 at r3](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-KyqEKuxq4Y6NE_LXgg3:-KyqXAnaOaxmcvJ4ROQg:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/e334e62e7e68f269f67366dd4320429c2689c1b1/drake/automotive/maliput/multilane/connection.h#L307)):*
<details><summary><i>Previously, EricCousineau-TRI wrote…</i></summary><blockquote>

BTW We only really need `{}` member declaration initialization for PODs that would not have it otherwise.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-Kyqb_Yk0rsnFBDrIKUg:bi0vpv0)*

**Comment by agalbachicar** [comment]
@EricCousineau-TRI failing tests have problems with Maven, should I re trigger them?

---

Review status: 3 of 5 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-Kyqu5DHNToqQmPJR6ib:b7cym1w)*

**Comment by EricCousineau-TRI** [comment]
Yup! Jamie put in a fix here, so re-triggering it should pull: https://github.com/RobotLocomotion/drake/pull/7465
(Just to be safe, I would rebase off of `upstream/master` and then force-push that.)

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-Kyqux02box2xCT_7Akc:b-fvo26q)*

**Comment by maddog-tri** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/7343#-:-KyrkQeph20P0djNDInX:bnfp4nl)*
